### PR TITLE
Rockchip: cleanup / fixup / update

### PR DESCRIPTION
**Pull request by knaerzche**
This is a further update/cleanup PR for Rockchip project.

Since there is no way to completly prevent from updates from earlier releases, I created a device tree mapping in `update.sh` script so that old ones will be replaced by the one that _suits_ best in mainline kernel (some HW might not work after the update for some devices, but at least the device will continue to boot/work with basic features). I've tested this for all 3 SoCs with combinations DT exists mainline / DT does not exist mainline.

I've also reduced the number of images that need be created: Only 1 image is build for every board (not for x revisions) and also dropped some boards where I found stuff we would need is missing in device tree. That probably could be fixed easy, but I have no HW to test them and they weren't supported in earlier releases anyway.

To be in line with other LE projects there are also some further addtions to the linux configs and also some fixups/addtions to upstream device trees that are needed to make all HW we may need work (namely ir-receiver and spdif for RK3328).

### CODE REVIEW

**Review by lrusak — COMMENTED** [review]
LGTM. The only thing I can add is the suggestion to use braces `{}` for variable substitution in the update.sh script. We didn't do this in the past but since the entire LE codebase was updated to use braces we should try and update the code that actually runs to use braces also. 👍

**Comment by knaerzche** [comment]
Done: converted all bootloader scripts to "braces" syntax using fixlecode.py and re-tested.

File: projects/Rockchip/devices/RK3288/linux/default/linux.arm.conf [hunk 1]
```diff
@@ -5780,7 +5780,7 @@ CONFIG_CRYPTO_LIB_SHA256=y
 CONFIG_CRYPTO_HW=y
 # CONFIG_CRYPTO_DEV_ATMEL_ECC is not set
 # CONFIG_CRYPTO_DEV_ATMEL_SHA204A is not set
-CONFIG_CRYPTO_DEV_ROCKCHIP=m
+# CONFIG_CRYPTO_DEV_ROCKCHIP is not set
```
Comment (jernejsk) [inline projects/Rockchip/devices/RK3288/linux/default/linux.arm.conf:5783]: why would you drop crypto driver?

Comment (knaerzche) [reply projects/Rockchip/devices/RK3288/linux/default/linux.arm.conf:5783]: see commit msg: "Using this driver crashes the SoC: drop it."

Comment (jernejsk) [reply projects/Rockchip/devices/RK3288/linux/default/linux.arm.conf:5783]: sorry, I missed it.

**Review by jernejsk — APPROVED** [review]

**Pull request merged**